src/Symfony/Component/Notifier/Notification/Notification.php

```diff
@@ -50,11 +50,14 @@ public function __construct(string $subject = '', array $channels = [])
         $this->channels = $channels;
     }
 
+    /**
+     * @return static
+     */
     public static function fromThrowable(\Throwable $exception, array $channels = []): self
     {
         $parts = explode('\\', \get_class($exception));
 
-        $notification = new self(sprintf('%s: %s', array_pop($parts), $exception->getMessage()), $channels);
+        $notification = new static(sprintf('%s: %s', array_pop($parts), $exception->getMessage()), $channels);
```

Comment:
This change assumes that the child class has the same constructor signature as its parent which is not necessarily the case.

Reply:
Indeed, I missed that. Should we then just add a method `exception`? Like this:
```php
    /**
     * @return $this
     */
    public function exception(\Throwable $exception): self
    {
        $parts = explode('\\', \get_class($exception));

        $this->subject = sprintf('%s: %s', array_pop($parts), $exception->getMessage());
        if (class_exists(FlattenException::class)) {
            $this->exception = $exception instanceof FlattenException ? $exception : FlattenException::createFromThrowable($exception);
        }
        $this->exceptionAsString = $this->computeExceptionAsString($exception);

        return $this;
    }
```

Reply:
That sounds like a good idea. But we would need to target 6.2 with that.

Reply:
@andersonamuller Your suggestion seems good to me. Can you close this PR and open another one on 6.2 with the new `exception()` method?